transports/tls/src/certificate.rs

```diff
@@ -43,9 +43,7 @@ static P2P_SIGNATURE_ALGORITHM: &rcgen::SignatureAlgorithm = &rcgen::PKCS_ECDSA_
 
 /// Generates a self-signed TLS certificate that includes a libp2p-specific
 /// certificate extension containing the public key of the given keypair.
-pub fn generate(
-    identity_keypair: &identity::Keypair,
-) -> Result<rcgen::Certificate, rcgen::RcgenError> {
+pub fn generate(identity_keypair: &identity::Keypair) -> Result<rcgen::Certificate, GenError> {
```

Comment:
Still `rcgen::Certificate` leaks?

Reply:
You are right, well spotted!

We can make this one `pub(crate)` though because it is only used by internal APIs.

Reply:
Resolved in [`e1f34ae` (#2945)](https://github.com/libp2p/rust-libp2p/pull/2945/commits/e1f34aeafb9a543465e52d6bfffdfeb620cd9843).